src/App.tsx

```diff
@@ -15,47 +15,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[], sortType : SortType, isReversed : boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((firstGood, secondGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return firstGood.localeCompare(secondGood);
+      case SortType.LENGTH:
+        return firstGood.length - secondGood.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, sortBy] = useState(SortType.NONE);
+  const [isReversed, reverseGoods] = useState(false);
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={sortType === SortType.ALPHABET
+            ? 'button is-info' : 'button is-info is-light'}
+          onClick={() => sortBy(SortType.ALPHABET)}
```

Comment:
Create separate functions for handling state change here and elsewhere.
```suggestion
          onClick={sortAlphabetically}
```